src/common/models/series/series.ts

```diff
@@ -29,15 +32,31 @@ export interface SeriesBehaviours {
   plywoodKey: (period?: SeriesDerivation) => string;
 }
 
-export type Series = MeasureSeries | ExpressionSeries;
+export type Series = MeasureSeries | ExpressionSeries | QuantileSeries;
 
-export function fromJS(params: any): Series {
+export function fromMeasure(measure: Measure): MeasureSeries | QuantileSeries {
+  if (measure.expression instanceof QuantileExpression) {
+    return QuantileSeries.fromQuantileMeasure(measure);
+  }
+  return MeasureSeries.fromMeasure(measure);
+}
+
+function fromJSWithMeasure(params: any, { expression }: Measure): MeasureSeries | QuantileSeries {
```

Comment:
`unknown`. !!!

Reply:
It is edge of the system. Somewhere you need any. Or unknown with full validation, which would be nice but out of scope for this task.